### ISSUE TITLE
Error http 405 when trying to import azurerm_notification_hub ressource

### ISSUE THREAD

**Issue by alex-3sr** [terraform-providers/terraform-provider-azurerm#7669]
Hi team,

I remember successfully in the past, to import some azurerm_notification_hub ressource.
Today, I encounter an issue for import this ressource type with this error : 

> Error: Error retrieving Credentials for Notification Hub "" (Namespace "notifications-xxx-yyy" / Resource Group "RG-XXX-NOTIF"): notificationhubs.Client#GetPnsCredentials: Failure responding to request: StatusCode=405 -- Original Error: autorest/azure: Service returned an error. Status=405 Code="HttpMethodIsNotSupported" Message="The HTTP method 'POST' is not supported."

I tried on multiple ressources, in different subscriptions, but still have this issue and can't import anymore some notification hub ressources.

Thanks for your help, for check if I'm alone in this case or not.
Regars
Alexandre

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
Terraform 12.26
Azure RM Provider 2.17

### Affected Resource(s)

* `azurerm_notification_hub ressource`

### Terraform Configuration Files

```hcl
resource "azurerm_notification_hub" "nh-xxx" {
  name                = lower("nh-${var.environment}-xxx")
  namespace_name      = azurerm_notification_hub_namespace.nh_namespace.name
  resource_group_name = azurerm_resource_group.rgproject.name
  location            = azurerm_resource_group.rgproject.location

  apns_credential {
    application_mode = "Production"
    bundle_id        = "com.xxx.yyy"
    key_id           = "ABCDEF"
    team_id          = "GHIJKL"
    token            = "MIGT...CNM"
  }
  gcm_credential {
    api_key = "AA...Un73"
  }

  tags                = local.common_tags
}
```

### Debug Output

azurerm_notification_hub.nh-xxx: Importing from ID "/subscriptions/xxx/resourceGroups/RG-xxx-NOTIF/providers/Microsoft.NotificationHubs/namespaces/notifications-xxx/NotificationHubs/notif-xxx-hub"...
azurerm_notification_hub.nh-xxx: Import prepared!
  Prepared azurerm_notification_hub for import
azurerm_notification_hub.nh-xxx: Refreshing state... [id=/subscriptions/3xxx/resourceGroups/RG-xxx-NOTIF/providers/Microsoft.NotificationHubs/namespaces/notifications-xxx-xxx/NotificationHubs/notif-xxx-hub]

Error: Error retrieving Credentials for Notification Hub "" (Namespace "notifications-xxx-xxxr" / Resource Group "RG-xxx-NOTIF"): notificationhubs.Client#GetPnsCredentials: Failure responding to request: StatusCode=405 -- Original Error: autorest/azure: Service returned an error. Status=405 Code="HttpMethodIsNotSupported" Message="The HTTP method 'POST' is not supported."

### Expected Behavior
Expected that Terraform import in state the ressource successfully

### Actual Behavior
The ressource is not imported

### Steps to Reproduce
1. Create config block azurerm_notification_hub
2. `terraform import -var-file="./Env/$target_env.tf-variables.tfvars" azurerm_notification_hub.nh-xxxe "/subscriptions/xxx/resourceGroups/RG-xxx-NOTIF/providers/Microsoft.NotificationHubs/namespaces/notifications-xxx-xxx/NotificationHubs/notif-xxx-hub"`

* #0000

**Comment by njuCZ** [comment]
@alex-3sr I am sorry that I could not reproduce this problem. Import notificationHub works find in my local

**Comment by alex-3sr** [comment]
> @alex-3sr I am sorry that I could not reproduce this problem. Import notificationHub works find in my local

Hi njuCZ, thanks for this repro. So surely something is bad on my side .. Can you just say me your tf and arm provider version please ?

thanks

**Comment by tombuildsstuff** [comment]
hi @alex-3sr 

Taking a look at the error message being returned here:

```
Error: Error retrieving Credentials for Notification Hub "" (Namespace "notifications-xxx-xxxr" / Resource Group "RG-xxx-NOTIF"): notificationhubs.Client#GetPnsCredentials: Failure responding to request: StatusCode=405 -- Original Error: autorest/azure: Service returned an error. Status=405 Code="HttpMethodIsNotSupported" Message="The HTTP method 'POST' is not supported."
```

It appears the "name" is missing from the request - we parse this out of the ID from the value for `notificationHubs` (rather than `NotificationHubs`, which is used in the example above) ([as per the documentation](https://www.terraform.io/docs/providers/azurerm/r/notification_hub.html#import)) - as such it should be possible to fix this by updating the casing of the ID here - would you be able to take a look and see if updating the case of this URI segment works for you?

We should look to add validation so this is clearer however - @njuCZ would you mind sending a PR to add a validator for this so it's clear which segment is missing?

Thanks!

**Comment by alex-3sr** [comment]
> hi @alex-3sr
> 
> Taking a look at the error message being returned here:
> 
> ```
> Error: Error retrieving Credentials for Notification Hub "" (Namespace "notifications-xxx-xxxr" / Resource Group "RG-xxx-NOTIF"): notificationhubs.Client#GetPnsCredentials: Failure responding to request: StatusCode=405 -- Original Error: autorest/azure: Service returned an error. Status=405 Code="HttpMethodIsNotSupported" Message="The HTTP method 'POST' is not supported."
> ```
> 
> It appears the "name" is missing from the request - we parse this out of the ID from the value for `notificationHubs` (rather than `NotificationHubs`, which is used in the example above) ([as per the documentation](https://www.terraform.io/docs/providers/azurerm/r/notification_hub.html#import)) - as such it should be possible to fix this by updating the casing of the ID here - would you be able to take a look and see if updating the case of this URI segment works for you?
> 
> We should look to add validation so this is clearer however - @njuCZ would you mind sending a PR to add a validator for this so it's clear which segment is missing?
> 
> Thanks!

:o you rocks, thank you @tombuildsstuff  ;)
You was right, by changing N to n in tf cmd import, now it's works like a charm !

Before (not working) -> terraform import -var-file="./Env/$target_env.tf-variables.tfvars" azurerm_notification_hub.nh-xxxe "/subscriptions/xxx/resourceGroups/RG-xxx-NOTIF/providers/Microsoft.NotificationHubs/namespaces/notifications-xxx-xxx/**NotificationHubs**/notif-xxx-hub"

Now (and working well) -> terraform import -var-file="./Env/$target_env.tf-variables.tfvars" azurerm_notification_hub.nh-xxxe "/subscriptions/xxx/resourceGroups/RG-xxx-NOTIF/providers/Microsoft.NotificationHubs/namespaces/notifications-xxx-xxx/**notificationHubs**/notif-xxx-hub"

In fact, I just copied ResourceId directly from GUI and it having this n as Upper case. In future I will not forget it again and will use in lower case. Maybe can we improve it by checking on validation when end-user start import with a help notice message about that ? Really, even if I'm not a novice in TF, I was really lost about this error and without your eye's expert, i wasn't unabme to fix it :)

Thanks,
Alex

**Comment by njuCZ** [comment]
@tombuildsstuff thanks for your pointing out the problem. I have submitted a PR to add the validation for ID.
@alex-3sr about importing a resource, it's better to refer to the document. Sorry for miss this case issue

**Comment by tombuildsstuff** [comment]
@alex-3sr glad to hear this is now working for you - we're working through adding validation to new resources (and retro-fitting it when we get a chance) - but as @njuCZ has mentioned we'd recommend using the ID format in the "import" section of the documentation - which should be up to date :)

In this instance this issue will be fixed by @njuCZ's PR (#7690) - which'll ship in the next version 👍
